lib/spatial/brute_force.h

```diff
@@ -0,0 +1,74 @@
+/// \file
+/// Maintainer: Felice Serena
+///
+
+#pragma once
+
+#include "spatial_oracle.h"
+#include <limits>
+#include <Eigen/Core>
+
+namespace MouseTrack {
+
+/// Reference implementation,
+/// performs queries in a naive, inefficient way
+/// Expects points to be Eigen col vectors of size Dim x 1
+/// The point list is a Dim x #Points eigen matrix
+
+template <typename _Precision, int _Dim>
+class BruteForce : public SpatialOracle<Eigen::Matrix<_Precision, _Dim, Eigen::Dynamic, Eigen::RowMajor + Eigen::AutoAlign>, Eigen::Matrix<_Precision, _Dim, 1>, _Precision> {
+public:
+    typedef Eigen::Matrix<_Precision, _Dim, Eigen::Dynamic, Eigen::RowMajor + Eigen::AutoAlign> PointList;
+    typedef Eigen::Matrix<_Precision, _Dim, 1> Point;
+private:
+    PointList _points;
```

Comment:
This is inefficient: Can we split the class to a safe and an unsafe version? the unsafe version would only hold a pointer to the data, lifetime management is responsibility of the client. The safe version would hold the data and also an instance of the unsafe version who's pointer is set to the internal copy. (maybe split to three classes to safe extra pointer in safe class?)

Reply:
push to optimization phase